packages/editor/src/hooks/pattern-partial-syncing.js

```diff
@@ -64,10 +64,44 @@ const withPartialSyncingControls = createHigherOrderComponent(
 	}
 );
 
+/**
+ * Adds an editOriginalPattern prop which allows the block to switch between post and pattern
+ * editing modes.
+ *
+ * @param {Component} BlockEdit Original component.
+ *
+ * @return {Component} Wrapped component.
+ */
+const withPatternOnlyRenderMode = createHigherOrderComponent(
+	( BlockEdit ) => ( props ) => {
+		const onSelectPost = useSelect(
+			( select ) =>
+				select( editorStore ).getEditorSettings().onSelectPost,
+			[]
+		);
+		if ( props.name !== 'core/block' ) {
+			return <BlockEdit { ...props } />;
+		}
+
+		const newProps = {
+			...props,
+			editOriginalPattern:
+				typeof onSelectPost === 'function' ? onSelectPost : undefined,
+		};
+
+		return <BlockEdit { ...newProps } />;
+	}
+);
```

Comment:
Both the site editor and post editor use the `editor` package, so I'm not sure that this code is in the right place given how it's post editor specific.

Reply:
We still want the `Edit original` button to appear in the site editor though don't we?